app/models/news_article.rb

```diff
@@ -20,8 +20,16 @@ class NewsArticle < Node
     order("content ->> 'name' ASC")
   }
 
+  scope :by_section, -> (section) {
+    where(:section_id => section.id)
+  }
+
   validates_presence_of :section
 
+  def self.published_for_section(section)
+    NewsArticle.by_section(section).by_release_date.by_published_at.published
```

Comment:
`by_release_date` and `by_published_at`. It seems odd to sort by both. The release date is the one the user can choose so that's probably the more relevant thing to sort by.